SConstruct

```diff
@@ -1135,7 +1139,10 @@ env.Append( CXXFLAGS = [ "-std=$CXXSTD" ] )
 if env["DEBUG"] :
 	env.Append( CXXFLAGS = [ "-g" ] )
 else :
-	env.Append( CXXFLAGS = [ "-DNDEBUG", "-DBOOST_DISABLE_ASSERTS" ] )
+	cxxFlags  =[ "-DNDEBUG", "-DBOOST_DISABLE_ASSERTS"]
+	if env["DEBUGINFO"] :
+		cxxFlags.append( "-g" )
```

Comment:
As far was I know it doesn't directly slow down execution speed but increases binary sizes.